integrated_channels/canvas/client.py

```diff
@@ -682,13 +683,34 @@ def _search_for_canvas_user_by_email(self, user_email):
         get_users_by_email_response = rsps.json()
 
         try:
-            canvas_user_id = get_users_by_email_response[0]['id']
+            canvas_user_id = get_users_by_email_response[0]["id"]
+            return canvas_user_id
         except (KeyError, IndexError) as error:
-            raise ClientError(
-                "No Canvas user ID found associated with email: {}".format(user_email),
-                HTTPStatus.NOT_FOUND.value
-            ) from error
-        return canvas_user_id
+            # learner is decommissioned on Canvas side - unlink it from enterprise
+            try:
+                enterprise_customer = self.enterprise_configuration.enterprise_customer
+                # Unlink user from related Enterprise Customer
+                EnterpriseCustomerUser.objects.unlink_user(
+                    enterprise_customer=enterprise_customer,
+                    user_email=user_email,
+                )
+                raise ClientError(
+                    "No Canvas user ID found associated with email: {} - User unlinked from enterprise now".format(
+                        user_email
+                    ),
+                    HTTPStatus.NOT_FOUND.value,
```

Comment:
this will continue to create noise in Splunk. Instead of raising client error, how about we just log it?

Reply:
I intentionally raised the error to break the flow otherwise any other error will be thrown by few other methods in client which are using this.

Also, this time noise will be **much** less because this error will be thrown **only** once for a learner. Once it's unlinked, we won't attempt its transmission in next job run.

Reply:
Makes sense!